modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -1745,10 +1672,18 @@ const DOMUtils = (doc: Document, settings: Partial<DOMUtilsSettings> = {}): DOMU
      * Returns true/false if the specified node is to be considered empty or not.
      *
      * @method isEmpty
-     * @param {Object} elements Optional name/value object with elements that are automatically treated as non-empty elements.
+     * @param {Node} node The node to check if it's empty.
+     * @param {Object} options An optional object with additional options.
+     *
+     * The options object can have the following properties:
+     * - skipBogus: If true, the function will skip over bogus nodes during the check. Optional, default is true.
+     * - includeZwsp: If true, the function will include zero-width space characters in the check. Optional, default is false.
+     * - check_root_as_content: If true, the function will check if the target node (instead of its children) is considered "content". Optional, default is false.
+     * - isContent: An optional function that takes a node as a parameter and returns a boolean indicating whether the node is considered "content".
```

Comment:
Do we need to expose all these at this time? Do we need that outside the core?
Also this breaks the old api since you can no longer specify the element names record. Also the old code would use that record instead of the set of that the `isNonEmptyElement` is now checking. 
I would suggest having a `isContentElement(el: Element)` or a `isNonEmptyElement(el: Element)` predicate option in the internal structure that would default to using the Empty `isNonEmptyElement` that checks the record string.
I wonder if this wasn't properly covered with tests. I wonder if we could possibly also handle the nbzwsp case with a `isNonEmptyTextNode` predicate internally? Predicates makes things way more flexible but one could always generate predicates from options or have predefined predicates that one can stick in.

- [ ] Restore the old API and see if we can make it compatible with the old elements behaviour

Reply:
I suspect the element names is not required now that we use the schema, but yes we need to be careful about breaking this API.

Reply:
I've updated the `DomUtils.isEmpty` function for compatibility with the old API.